vortex-file/src/read/buffered.rs

```diff
@@ -26,11 +26,11 @@ pub(crate) trait ReadMasked {
 
 /// Read an array with a [`RowMask`].
 pub(crate) struct ReadArray {
-    layout: Box<dyn LayoutReader>,
+    layout: Arc<dyn LayoutReader>,
```

Comment:
So LayoutReaders are no longer stateful?

Reply:
ok after a more careful reading and a small test, as far as I can tell while the `LayoutReader` implementations are stateful (with interior mutability), they can read independently from each other as the `LayoutBufferedReader` is the part that actually controls the overall flow and each stream initializes a new one.

Reply:
There can be multiple concurrent reads being tracked in the layout reader but are they reusable?

I thought something encoded assumptions that splits were accessed monotonically but maybe that was in the BufferedReader

Reply:
I thought so too, but that does seem to be in the `BufferedLayoutReader`. The `LayoutReader` implementation read through `poll_read` which takes a `RowMask` to read try and read a specific range on each call, and the `BufferedReader` tracks that.

Reply:
You should be able to access the layoutreaders out of order so we can read multiple at the same time irrespective of their children.